app/src/tests/components/contents/PacketReadPermission/utils/getRolesUsersWithOnlyReadPacketPermission.test.ts

```diff
@@ -0,0 +1,91 @@
+import { RoleWithRelationships } from "../../../../../app/components/contents/manageAccess/types/RoleWithRelationships";
+import { UserWithRoles } from "../../../../../app/components/contents/manageAccess/types/UserWithRoles";
+// eslint-disable-next-line max-len
+import { getRolesUsersWithOnlyReadPacketPermission } from "../../../../../app/components/contents/PacketReadPermission/utils/getRolesUsersWithOnlyReadPacketPermission";
+
+const mockMapPermissionsToNames = jest.fn().mockReturnValue([]);
+const mockHasPacketReadPermissionForPacket = jest.fn();
+const mockCanReadPacketGroup = jest.fn();
+const mockCanManagePacket = jest.fn();
+jest.mock("../../../../../lib/constructPermissionName", () => ({
+  mapPermissionsToNames: () => mockMapPermissionsToNames()
+}));
+jest.mock("../../../../../lib/auth/hasPermission", () => ({
+  hasPacketReadPermissionForPacket: () => mockHasPacketReadPermissionForPacket(),
+  canReadPacketGroup: () => mockCanReadPacketGroup(),
+  canManagePacket: () => mockCanManagePacket()
+}));
```

Comment:
Because all of these permission utils are quite tightly dependent on each other  I wonder if we shouldn't mock these `hasPermission` methods here. Like, I was wondering if we should test the case here of a user with global read permission... which of course should be covered by `canReadpacketGroup`, but that's being mocked.. Maybe a lot of this is going away in the refactor to the back end though, so it might not be worth it!

Reply:
yeah good idea!! il keep that in mind when testing the refactor code.... i wont change here as this function will be removed in refactor